KCheckbox: Remove local state management, rely on given props

nucleogenesis opened the pull request:
<!-- Please remove any unused sections -->

## Description
<!-- What does this PR do? Briefly describe in 1-2 sentences* -->

KCheckbox had local data which are initialized to the given props for `indeterminate` and `checked` and then updated in a watcher. This resulted in that clicking a checkbox would always toggle it's `isCurrentlyChecked` value and setting `isCurrentlyIndeterminate` to false.

The changes here instead have KCheckbox simply use the given prop values and does nothing to manage any internal state around whether it shows `indeterminate` and/or `checked`.

This results in users of KCheckbox needing to be mindful of how they manage the values they pass into KCheckbox's props as the component will now reflect their values at all time.

#### Issue addressed

This permits Studio's clipboard to have the ability to tell KCheckbox what it should be looking like.



## Changelog
<!-- [DO NOT REMOVE-USED BY GH ACTION] CHANGELOG START -->
<!--
  - Fill in the changelog item(s) below. If there are more groups of closely
    related changes, prepare more changelog items for each one of them.
    At a minimum, always separete non-breaking changes from breaking changes.
  - This needs to be pasted to CHANGELOG.md before merging a PR.
  - See changelog guidelines https://www.notion.so/learningequality/DRAFT-Changelog-Guidelines-106b6ebbdeda4ba5b3b3e7c490c5a4fe and existing
    items in CHANGELOG.md as examples
 -->

  - **Description:** Removes internal state management for checked & indeterminate in KCheckbox.
  - **Products impact:** Updated API
  - **Addresses:** https://github.com/learningequality/studio/issues/4636
  - **Components:** KCheckbox
  - **Breaking:** No
  - **Impacts a11y:** No
  - **Guidance:** If you use KCheckbox, it is your responsibility to handle the `change` event and update whether or not the given `checked` and `indeterminate` props reflect the reality that you expect.

<!-- [DO NOT REMOVE-USED BY GH ACTION] CHANGELOG END -->

## Steps to test

Click all of the checkboxes and the states of indeterminacy vs checked-ness -- particularly in nested situations.

The most common in Kolibri is content selection. In Studio, this issue was found while debugging a problem with the Clipboard.

KOLIBRI:

- Resource selection during import
- During Quiz creation
- During Lesson creation

STUDIO:

- Clipboard
- Channel editor


## (optional) Implementation notes

### At a high level, how did you implement this?
<!-- Briefly describe how this works -->

### Does this introduce any tech-debt items?
<!-- List anything that will need to be addressed later -->

## Testing checklist
<!-- Complete the checklist before submitting a PR; delete anything that doesn't apply -->

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical and brittle code paths are covered by unit tests
- [ ] The change is described in the changelog section above

## Reviewer guidance
<!-- Delete anything that doesn't apply so your reviewer knows what to check for -->

- [ ] Is the code clean and well-commented?
- [ ] Are there tests for this change?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?
- [ ] _Add other things to check for here_

## After review

- [ ] The changelog item has been pasted to the `CHANGELOG.md`

## Comments
<!-- Any additional notes you'd like to add -->

AlexVelezLl commented:
Code changes make sense to me. But I wonder if removing the local state management is a breaking change.

I would suggest to make a different PR to remove the internal state targeting to develop and explore there its implications. And leave this PR with the removal of the `this.isCurrentlyIndeterminate = false;` line when we are toggling the check, which I suspect would be the needed change to let the consumer have total control in the combination of the four possible combinations of checked/unchecked/indeterminate/non-indeterminate states (as we have a watcher for the `checked` and `indeterminate` states.

For example, this change breaks the docs page as it relies on the KCheckbox internal state:

https://github.com/user-attachments/assets/33b05a26-be8e-45cf-9481-9b541399675f

And there could be some other places where we could be relying on the KCheckbox internal state too (also because the `checked` prop isnt even marked as required).

MisRob commented on CHANGELOG.md:
Leaving a note here that before merging it'd be good to revert CHANGELOG.md changes since the new GH action will now update the CHANGELOG.md after merging this PR by pasting the changelog from the PR description. Otherwise we'd have duplicate entries.

nucleogenesis committed:
KCheckbox: Remove local state management, rely on given props

    KCheckbox had local data which are initialized to the given props for
    `indeterminate` and `checked` and then updated in a watcher. This
    resulted in that clicking a checkbox would always toggle it's
    `isCurrentlyChecked` value and setting `isCurrentlyIndeterminate` to
    false.

    The changes here instead have KCheckbox simply use the given prop values
    and does nothing to manage any internal state around whether it shows
    `indeterminate` and/or `checked`.

    This results in users of KCheckbox needing to be mindful of how they
    manage the values they pass into KCheckbox's props as the component will
    now reflect their values at all time

AlexVelezLl reviewed (approved):
lgtm!

nucleogenesis committed:
kcheckbox docs updated to use props rather than rely on internal state

AlexVelezLl commented:
Thank you @nucleogenesis!
